soroban-sdk/src/env.rs

```diff
@@ -439,26 +436,41 @@ impl Env {
     ///
     /// # fn main() {
     /// let env = Env::default();
-    /// let contract_id = BytesN::from_array(&env, &[0; 32]);
-    /// env.register_contract_wasm(&contract_id, WASM);
+    /// env.install_contract_wasm(WASM);
     /// # }
     /// ```
-    pub fn register_contract_wasm<'a>(
-        &self,
-        contract_id: impl Into<Option<&'a BytesN<32>>>,
-        contract_wasm: &[u8],
-    ) -> BytesN<32> {
-        let contract_id = if let Some(contract_id) = contract_id.into() {
-            contract_id.clone()
-        } else {
-            let mut contract_id = [0u8; 32];
-            rand::thread_rng().fill_bytes(&mut contract_id);
-            BytesN::from_array(self, &contract_id)
-        };
+    pub fn install_contract_wasm<'a>(&self, contract_wasm: &[u8]) -> BytesN<32> {
```

Comment:
This lifetime looks unused.
```suggestion
    pub fn install_contract_wasm(&self, contract_wasm: &[u8]) -> BytesN<32> {
```

Reply:
Removed.